packages/db/src/abstractRepository.ts

```diff
@@ -124,35 +145,41 @@ export abstract class Repository<I extends Id, T> {
   }
 
   async keys(opts?: IFilterOptions<I>): Promise<I[]> {
+    this.dbReadsMetrics?.inc();
     const data = await this.db.keys(this.dbFilterOptions(opts));
     return (data || []).map((data) => this.decodeKey(data));
   }
   async *keysStream(opts?: IFilterOptions<I>): AsyncIterable<I> {
+    this.dbReadsMetrics?.inc();
     const keysStream = this.db.keysStream(this.dbFilterOptions(opts));
     const decodeKey = this.decodeKey.bind(this);
     for await (const key of keysStream) {
       yield decodeKey(key);
     }
   }
   async values(opts?: IFilterOptions<I>): Promise<T[]> {
+    this.dbReadsMetrics?.inc();
     const data = await this.db.values(this.dbFilterOptions(opts));
     return (data || []).map((data) => this.decodeValue(data));
   }
   async *valuesStream(opts?: IFilterOptions<I>): AsyncIterable<T> {
+    this.dbReadsMetrics?.inc();
```

Comment:
Here we are counting 1 for potentially reading the entire DB. However for writes we count 2 if writing two items. Should we just count 1 for batch writes? Level DB optimizes batch write opts so maybe our metrics should help us optimize DB usage in the direction, rewarding using batch writes vs one-off writes

Reply:
How would you count it since you don't know how many results we got though. I feel like this should be separate metrics like number of read streames opened.

Reply:
What you did for reads is fine. I would change for writes to always inc(1) without counting the num of items in the batch